cpp/sopt/imaging_forward_backward.h

```diff
@@ -316,7 +317,7 @@ bool ImagingForwardBackward<SCALAR>::objective_convergence(mpi::Communicator con
   if (scalvar.relative_tolerance() <= 0e0) return true;
   auto const current = obj_comm.all_sum_all<t_real>(
 	((gamma() > 0) ? g_proximal_->proximal_norm(x)
-       * gamma() : 0) + std::pow(sopt::l2_norm(residual), 2) / (2 * sigma() * sigma()));
```

Comment:
Both `gamma` and `sigma` can be accessed by the private `gamma_` variable or the public `gamma()` getter. I suppose the difference is the getter returns a `const` reference. Was there a reason for changing from `sigma()` to `sigma_` here?

Reply:
I think it was a change I made as I was trying to go through and understand / clarify the code to myself. It makes no difference to the calculation, but going through a getter method _inside the class you're getting it from_ seems utterly bizarre and serves no actual purpose, and makes the code much harder to read and understand.

Reply:
The unnecessary function calls may also end up having an adverse effect on performance depending on what optimisation ends up happening at compile, and currently the code makes function calls for almost every single variable access.

Reply:
I can put it back for consistency because I did put most of the variables that I played with back, but we might want to think at some point about whether this style is a good strategy.